lib/auth/auth_with_roles.go

```diff
@@ -1822,9 +1832,12 @@ func (r resourceChecker) CanAccess(resource types.Resource) error {
 		}
 	case types.SAMLIdPServiceProvider:
 		return r.CheckAccess(rr, state)
+
+	case services.UnifiedResource153Adapter[services.IdentityCenterAccount]:
```

Comment:
Why is this different from other cases above i.e. why aren't we matching on `types.KindIdentityCenterAccount` here?